tests/unit/src/util/Git.ts

```diff
@@ -0,0 +1,350 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import loadModule, { cleanupModuleMocks } from '../../../_support/loadModule';
+import { stub, SinonStub } from 'sinon';
+import Git from '../../../../src/util/Git';
+import * as env from '../../../../src/util/environment';
+
+let Module: any;
+let git: Git;
+let promiseSpawnStub: SinonStub;
+let promiseExecStub: SinonStub;
+let execStub: SinonStub;
+let loggerStub: SinonStub;
+let toStringStub: SinonStub;
+let existsSyncStub: SinonStub;
+let chmodSyncStub: SinonStub;
+
+registerSuite({
+	name: 'util/Git',
+
+	before() {
+		promiseSpawnStub = stub();
+		promiseExecStub = stub();
+		execStub = stub();
+		loggerStub = stub();
+		toStringStub = stub();
+		existsSyncStub = stub();
+		chmodSyncStub = stub();
+	},
+
+	after() {
+		cleanupModuleMocks();
+	},
+
+	beforeEach() {
+		Module = loadModule('src/util/Git', {
+			'./process': {
+				promiseSpawn: promiseSpawnStub,
+				promiseExec: promiseExecStub,
+				exec: execStub
+			},
+			'../log': {
+				logger: {
+					info: loggerStub
+				}
+			},
+			'./streams': {
+				toString: toStringStub
+			},
+			fs: {
+				existsSync: existsSyncStub,
+				chmodSync: chmodSyncStub
+			}
+		});
+
+		git = new Module();
+	},
+
+	afterEach() {
+		promiseSpawnStub.reset();
+		promiseExecStub.reset();
+		execStub.reset();
+		loggerStub.reset();
+		toStringStub.reset();
+		existsSyncStub.reset();
+		chmodSyncStub.reset();
+	},
+
+	'constructor': {
+		'with params'() {
+			const gitWithArgs = new Module('dir', 'file');
+			assert.equal(gitWithArgs.cloneDirectory, 'dir');
+			assert.equal(gitWithArgs.keyFile, 'file');
+		},
+		'default params'() {
+			assert.equal(git.cloneDirectory, process.cwd());
+			assert.equal(git.keyFile, env.keyFile());
+		}
+	},
+
+	async add() {
+		promiseExecStub.withArgs('git add file1 file2', {
+			silent: false,
+			cwd: git.cloneDirectory
+		}).returns('pass');
+
+		const actual = git.add('file1', 'file2');
+
+		assert.instanceOf(actual, Promise);
+		assert.strictEqual(await actual, 'pass');
+	},
+
+	async assert() {
+		git.isInitialized = () => false;
+		git.assert('url').then(() => assert.fail(), (error: Error) => {
+			assert.strictEqual(error.message,
+				`Repository is not initialized at "${ git.cloneDirectory }"`);
+		});
+
+		git.isInitialized = () => true;
+		git.getConfig = stub().withArgs('remote.origin.url')
+			.returns('url');
+
+		git.assert('other_url').then(() => assert.fail(), (error: Error) => {
+			assert.strictEqual(error.message,
+				'Repository mismatch. Expected "url" to be "other_url".');
+		});
+		assert.doesNotThrow(async () => await git.assert('url'));
```

Comment:
I don't think doesNotThrow wrappers like this work with async yet. Once async/await isn't transpiled it should be ok.

can we switch it to `return git.assert('url');`